addons/web_unsplash/static/src/js/unsplash_beacon.js

```diff
@@ -22,6 +22,16 @@ publicWidget.registry.UnsplashBeacon = publicWidget.Widget.extend({
                 if (!appID) {
                     return;
                 }
+                // Test assets JS is not loaded before this beacon is started,
+                // there is no way to guarantee that it will be loaded before
+                // the appID is obtained. Therefore this testing-purpose patch
+                // needs to be done within this file.
+                if (window.location.search.includes("test_unsplash_beacon")) {
+                    for (const imageEl of document.querySelectorAll(`img[src^="/unsplash/"]`)) {
+                        imageEl.dataset.beacon = "sent";
+                    }
+                    return;
+                }
```

Comment:
Let's try to do it inside a `<script>` inside the `arch` - maybe this would work. (Idea from @xO-Tx)